### PR TITLE
Import OWL classes and properties from Turtle files

### LINKED ISSUE #69: Full ontology Turtle import
## Context

Part of #50 (Vocabulary Mapping Interface)

The taxonomy builder needs to import complete lightweight ontologies from Turtle files, including classes, properties, and concept schemes. This enables importing the EEF ontology for mapping.

### Parse Turtle Files
Import RDF/Turtle files containing:
- `owl:Class` declarations
- `owl:ObjectProperty` and `owl:DatatypeProperty` declarations
- `skos:ConceptScheme` and `skos:Concept` (existing capability)

### Class Import
Extract from Turtle:
- Class URI, label (`rdfs:label`), description (`rdfs:comment`), scope note (`skos:scopeNote`)

### Property Import
Extract from Turtle:
- Property URI, label, description
- Domain class (`rdfs:domain`)
- Range - either a concept scheme reference or datatype (`rdfs:range`)
- Cardinality hints (e.g., `eef:allowMultiple`)
- Source metadata (e.g., `eef:eppiQuestion`) - store in description or dedicated field

### Concept Scheme/Concept Import
Extend existing import to work alongside class/property import:
- Schemes with their concepts
- Hierarchy (broader/narrower via `skos:broader`, `skos:narrower`, `skos:hasTopConcept`)
- Labels, definitions, notations

### Wire Up References
After import:
- Properties reference their domain classes (by URI)
- Properties with object ranges reference their schemes (by URI or FK if same project)

### Import Creates Owned Content
Imported content belongs to and is editable within the target project. This is different from referencing published external content.

## Use Case

Import `eef-ontology.ttl` and `eef-vocabulary.ttl` to create a complete EEF project with:
- 2 classes (Reference, Outcome)
- ~70 properties (object and datatype)
- 29 concept schemes with ~216 concepts

## Dependencies

- #66 - OWL Class support backend

## Part of

#50 - Vocabulary Mapping Interface

### PR DESCRIPTION
**Pull request by jaybea**
## Summary

Extends the SKOS Turtle import to handle full lightweight ontologies — OWL classes,
object/datatype properties, and their range references — alongside existing concept
scheme/concept import.

- Parse `owl:Class`, `owl:ObjectProperty`, and `owl:DatatypeProperty` from Turtle files
- Resolve property ranges to concept schemes, OWL classes, or XSD datatypes
- `range_class` is a URI string on Property (matching `domain_class` pattern) — no FK to `ontology_classes`
- Enforce exactly-one-of `range_scheme_id`/`range_datatype`/`range_class` at both schema and service level
- Parse `allowMultiple` annotation → `cardinality` field (namespace-agnostic)
- Unresolved or ambiguous range URIs are stored as-is with a warning (ambiguous = instances span multiple schemes)
- Cross-file import: ranges resolve against existing project data from prior imports
- Idempotent: duplicate schemes, classes, and properties are silently skipped (by URI, with identifier fallback)
- Domain-less properties (`rdfs:domain` omitted) are skipped with warning
- Deduplicate dual-typed properties (both `owl:ObjectProperty` and `owl:DatatypeProperty`) by resolving type from `rdfs:range`
- Frontend: display class/property counts in import preview, show warnings

## Test plan
- [x] 639 backend tests pass
- [x] Manual: imported `eef-ontology.ttl` + `eef-vocabulary.ttl` — verified `allowMultiple` maps correctly (3 multiple, 4 single)
- [x] Manual: imported `eef-test-ontology.ttl` + `eef-test-vocabulary.ttl` — 70 properties, 23 multiple / 47 single (exact match to source)
- [x] Manual: imported `esea-vocab.ttl` — 11 schemes, 123 concepts, 12 properties with cross-file range resolution
- [x] Manual: imported `evrepo-core.ttl` — 8 classes, 14 properties (1 warning for `skos:Concept` range)
- [x] Manual: imported `ceds-ontology.ttl` — 943 schemes, 19,489 concepts, 942 classes
- [x] Manual: imported `eef-test-pathological.ttl` — edge cases handled with appropriate warnings

## Migration

Single migration adds `range_class` (varchar 2048, nullable) to `properties`. Includes a merge migration to resolve parallel heads from main.

## Known limitations
- `skos:narrower` and `skos:hasTopConcept` not parsed (only `skos:broader`, `skos:inScheme`, `skos:topConceptOf`)
- `eef:eppiQuestion` and other custom annotation properties not stored (no model field)
- `OntologyClass.uri` and `Property.uri` are computed from project namespace — imported `domain_class` URIs store the Turtle-native URI, so internal wiring assumes namespaces match. #97 stores URI as a column and fixes this.
- Imported classes/properties are not yet visible in the project UI sidebar. Tracked in #93.

Closes #69

### CODE REVIEW

File: backend/src/taxonomy_builder/schemas/property.py [hunk 1]
```diff
@@ -91,6 +107,20 @@ def validate_datatype(cls, v: str | None) -> str | None:
             raise ValueError(f"range_datatype must be one of: {allowed}")
         return v
 
+    @model_validator(mode="after")
+    def check_at_most_one_range(self) -> "PropertyUpdate":  # noqa: UP037
+        """Reject updates that set multiple range fields simultaneously."""
```
[the remark this answers is not in the document]
Comment (jaybea) [reply backend/src/taxonomy_builder/schemas/property.py]: Extracted!

File: backend/src/taxonomy_builder/services/skos_import_service.py [hunk 2]
```diff
@@ -314,59 +557,199 @@ async def preview(
...
+            if pm["range_uri"] and pm["property_type"] == "object":
+                resolved = self._resolve_object_range(
+                    g, pm["range_uri"], scheme_uris, class_uris
+                )
+                if resolved and resolved[0] == "scheme":
+                    range_scheme_title = scheme_uri_to_title.get(resolved[1])
+                elif resolved is None:
+                    warnings.append(
+                        f"Property '{pm['identifier']}' range '{pm['range_uri']}' "
+                        f"not found in project"
```
Comment (danielbreves) [inline backend/src/taxonomy_builder/services/skos_import_service.py]: Should we say not found or ambiguous scheme here?

File: backend/src/taxonomy_builder/services/skos_import_service.py [hunk 3]
```diff
@@ -242,69 +416,138 @@ def _analyze_graph(
...
+            valid=True,
+            schemes=scheme_previews,
+            total_concepts_count=total_concepts,
+            total_relationships_count=total_relationships,
+            classes=class_previews,
+            properties=property_previews,
+            classes_count=len(class_previews),
+            properties_count=len(property_previews),
+            warnings=warnings,
+            errors=analysis["warnings"],
```
Comment (danielbreves) [inline backend/src/taxonomy_builder/services/skos_import_service.py]: Should we merge `warnings` with `analysis["warnings"]` perhaps instead of passing as `errors`?

File: backend/tests/test_models/test_concept_broader.py [hunk 4]
```diff
@@ -104,24 +103,6 @@ async def test_concept_can_be_broader_to_multiple(
     assert len(narrower_rels) == 2
 
 
-@pytest.mark.asyncio
-async def test_duplicate_broader_relationship_fails(
```
Comment (danielbreves) [inline backend/tests/test_models/test_concept_broader.py]: Just wondering what prompted removing this test?
